text/trace/0168-sampling-propagation.md

```diff
@@ -0,0 +1,181 @@
+these probabilities as small integer values using the base-2 logarithm
+of the adjusted count (i.e., inverse probability).
+
+For example, the probability value 2 corresponds with 1-in-4 sampling,
+the probability value 10 corresponds with 1-in-1024 sampling.  Using
+one byte of information we can convey sampling rates as small as 2^-255.
+
+### Random value
+
+With head trace sampling probabilities limited to powers of two, the
+amount of randomness needed per trace context is limited.  A
+consistent sampling decision is accomplished by propagating a
+geometrically distributed random variable with shape parameter `1/2`,
+requiring only two bits of randomness on average per trace.  See
+[Estimation from Partially Sampled Distributed
+Traces](https://arxiv.org/pdf/2107.07703.pdf) section 2.8 for a
+detailed explanation.
+
+Such a random variable `r` can be generated using the following
+pseudocode:
+
+```
+r := 0
+for {
+  if nextRandomBit() {
+    break // The expected value of r is 2
+  }
+  r++
+}
+```
+
+This can be computed from a stream of random bits as the number of
+leading zeros using efficient instructions on modern computer
+architectures.
+
+For example, the value 3 means there were three leading zeros and
+corresponds with being sampled at probabilities 1-in-1 through 1-in-8
+but not at probabilities 1-in-16 and smaller.  Using one byte of
+information we can convey a consistent sampling decision for sampling
+rates as small as 2^-255.
+
+### Proposed `tracestate` syntax
+
+The consistent sampling decision and head trace sampling probability
+will be propagated using four bytes of base16 content, as follows:
+
+```
+tracestate: otelprob=PPRR
+```
+
+where `PP` are two bytes of base16 probability value and `RR` are two
+bytes of base16 random value.
+
+### Examples
+
+The following `tracestate` value:
+
+```
+tracestate: otelprob=0a03
+```
+
+translates to
+
+```
+base16(probability) = 03 // 1-in-8 head probability
+base16(randomness) = 0a // qualifies for 1-in-1024 sampling or greater
+```
+
+Any `TraceIDRatioBased` Sampler configured with probability 2^-10 or
+greater will enable sampling this trace, whereas any
+`TraceIDRatioBased` Sampler configured with probability 2^-11 or less
+will stop sampling this trace.  The W3C `sampled` flag is set to true
+when the probability value is less than or equal to the randomness
+value.
+
+## Internal details
+
+The reasoning behind restricting the set of sampling rates is that it:
+
+- Lowers the cost of propagating head sampling probability
+- Limits the number of random bits required
+- Makes math involving partial traces tractable.
+
+[An algorithm for making statistical inference from partially-sampled
+traces has been published](https://arxiv.org/pdf/2107.07703.pdf) that
+explains how to work with a limited number of power-of-2 sampling rates.
+
+### Behavior of the `TraceIDRatioBased` Sampler
+
+The Sampler must be configured with a power-of-two probability
+`P=2^-S`.
```

Comment:
Defined limit of `S<=62` per @oertl comment?

Reply:
I've changed the text to refer to "6 bits" of randomness, with 63 reserved to mean zero probability.